lib/authorizer/basic.js

```diff
@@ -43,7 +44,7 @@ module.exports = {
      * @todo - this should be made into ".sign" after the signature calculation is moved out of the SDK.
      */
     sign: function (params, request) {
-        if (!params || !params.username) { return request; } // Nothing to do if no parameters are present.
+        if (!params || !_.has(params, 'username') || !_.has(params, 'password')) { return request; }
```

Comment:
This condition will still cause a bailout if the username is not present. Since username is not meant to be required anymore, you'd want to check for the password exclusively, like so:
```js
if (!(params && _.has(params, 'password'))) { return request; }
```

Reply:
No no no, this does not allow the username to be absent
Only to be empty string (same for password, it can be empty string)
I will change the title to be more precise